### ISSUE TITLE
Accuracy after training by default settings is lower than paper

### ISSUE THREAD

**Issue by xmlyqing00** [smhassanerfani/atlantis#11]
Hi, 

I used the pretrained model `snapshots/resnet101-imagenet.pth` and the default training settings to train the model your dataset. They converged well to ACC 53.5%, but it is lower than the number of Table 3 in the paper, which is ACC = 75.18%. Do you have any ideas what should I tune the model or parameters? 

Thanks

> 
Epoch 27
-------------------------------
loss: 1.36548, lr = 0.000040 [ 87664/100920]
loss: 1.03685, lr = 0.000040 [ 87864/100920]
loss: 0.63693, lr = 0.000039 [ 88064/100920]
loss: 0.78273, lr = 0.000039 [ 88264/100920]
loss: 0.75318, lr = 0.000038 [ 88464/100920]
loss: 0.89742, lr = 0.000037 [ 88664/100920]
loss: 0.43634, lr = 0.000037 [ 88864/100920]
loss: 0.39186, lr = 0.000036 [ 89064/100920]
loss: 2.59725, lr = 0.000036 [ 89264/100920]
loss: 0.55175, lr = 0.000035 [ 89464/100920]
loss: 1.37644, lr = 0.000035 [ 89664/100920]
loss: 0.48361, lr = 0.000034 [ 89864/100920]
loss: 0.52164, lr = 0.000034 [ 90064/100920]
loss: 1.09766, lr = 0.000033 [ 90264/100920]
loss: 1.06286, lr = 0.000032 [ 90464/100920]
loss: 0.84704, lr = 0.000032 [ 90664/100920]
Test Error: 
 Accuracy: 54.0%, Avg loss: 1.435948 

Epoch 28
-------------------------------
loss: 0.75643, lr = 0.000031 [ 91028/100920]
loss: 0.87581, lr = 0.000030 [ 91228/100920]
loss: 0.74861, lr = 0.000030 [ 91428/100920]
loss: 0.77622, lr = 0.000029 [ 91628/100920]
loss: 1.11455, lr = 0.000029 [ 91828/100920]
loss: 0.30476, lr = 0.000028 [ 92028/100920]
loss: 0.63295, lr = 0.000028 [ 92228/100920]
loss: 1.81328, lr = 0.000027 [ 92428/100920]
loss: 1.04307, lr = 0.000026 [ 92628/100920]
loss: 0.59720, lr = 0.000026 [ 92828/100920]
loss: 1.40038, lr = 0.000025 [ 93028/100920]
loss: 1.06145, lr = 0.000025 [ 93228/100920]
loss: 1.16329, lr = 0.000024 [ 93428/100920]
loss: 0.93909, lr = 0.000023 [ 93628/100920]
loss: 0.69800, lr = 0.000023 [ 93828/100920]
loss: 0.57154, lr = 0.000022 [ 94028/100920]
Test Error: 
 Accuracy: 53.1%, Avg loss: 1.379513 

Epoch 29
-------------------------------
loss: 2.20924, lr = 0.000021 [ 94392/100920]
loss: 0.50997, lr = 0.000021 [ 94592/100920]
loss: 0.70862, lr = 0.000020 [ 94792/100920]
loss: 0.63156, lr = 0.000019 [ 94992/100920]
loss: 0.87382, lr = 0.000019 [ 95192/100920]
loss: 1.04872, lr = 0.000018 [ 95392/100920]
loss: 0.63661, lr = 0.000018 [ 95592/100920]
loss: 0.55542, lr = 0.000017 [ 95792/100920]
loss: 0.62491, lr = 0.000017 [ 95992/100920]
loss: 0.97411, lr = 0.000016 [ 96192/100920]
loss: 0.48036, lr = 0.000015 [ 96392/100920]
loss: 0.52839, lr = 0.000015 [ 96592/100920]
loss: 0.71433, lr = 0.000014 [ 96792/100920]
loss: 1.44972, lr = 0.000013 [ 96992/100920]
loss: 1.65513, lr = 0.000013 [ 97192/100920]
loss: 0.78470, lr = 0.000012 [ 97392/100920]
Test Error: 
 Accuracy: 53.9%, Avg loss: 1.387713 

Epoch 30
-------------------------------
loss: 0.35672, lr = 0.000011 [ 97756/100920]
loss: 2.15150, lr = 0.000010 [ 97956/100920]
loss: 0.76016, lr = 0.000010 [ 98156/100920]
loss: 1.47951, lr = 0.000009 [ 98356/100920]
loss: 0.86753, lr = 0.000009 [ 98556/100920]
loss: 1.48480, lr = 0.000008 [ 98756/100920]
loss: 0.43394, lr = 0.000007 [ 98956/100920]
loss: 1.16021, lr = 0.000007 [ 99156/100920]
loss: 0.40870, lr = 0.000006 [ 99356/100920]
loss: 1.20017, lr = 0.000005 [ 99556/100920]
loss: 0.38307, lr = 0.000005 [ 99756/100920]
loss: 0.49153, lr = 0.000004 [ 99956/100920]
loss: 1.18909, lr = 0.000003 [100156/100920]
loss: 0.46353, lr = 0.000002 [100356/100920]
loss: 1.53238, lr = 0.000002 [100556/100920]
loss: 0.47965, lr = 0.000001 [100756/100920]
Test Error: 
 Accuracy: 53.5%, Avg loss: 1.396503 

Done!

**Comment by smhassanerfani** [comment]
I found that you presented the results on the validation set, while the paper reported results are on the testing set.

**Comment by xmlyqing00** [comment]
Yeah, I see. I tried the `test.py`, it works. Do you have a plan to release your metric code in the test.py? Because now it only has dataloader and model to produce the segmentation, but doesn't produce the accuracy scores. That would be a great help!

Appreciate!

**Comment by smhassanerfani** [comment]
we have compute_iou.py in utils which might help. I put the link below.
https://github.com/smhassanerfani/atlantis/blob/10c413195c9405460fbf6acb0bf3510821a3364c/utils/compute_iou.py#L1

**Comment by xmlyqing00** [comment]
I see. Thank you!!
